internal/storage/pmem/pmem.go

```diff
@@ -44,10 +48,84 @@ func Mount(ctx context.Context, device uint32, target string) (err error) {
 			osRemoveAll(target)
 		}
 	}()
-	source := fmt.Sprintf("/dev/pmem%d", device)
+	source := fmt.Sprintf(pMemFmt, device)
 	flags := uintptr(unix.MS_RDONLY)
 	if err := unixMount(source, target, "ext4", flags, "noload"); err != nil {
 		return errors.Wrapf(err, "failed to mount pmem device %s onto %s", source, target)
 	}
 	return nil
 }
+
+// MountDM creates dm-linear block device and mounts it on `target`
+func MountDM(ctx context.Context, device uint32, deviceStart, deviceSize int64, target string) (err error) {
+	_, span := trace.StartSpan(ctx, "pmem::MountDM")
+	defer span.End()
+	defer func() { oc.SetSpanStatus(span, err) }()
+
+	span.AddAttributes(
+		trace.Int64Attribute("device", int64(device)),
+		trace.Int64Attribute("deviceStart", deviceStart),
+		trace.Int64Attribute("sectorSize", deviceSize),
+		trace.StringAttribute("target", target))
+
+	if err := osMkdirAll(target, 0700); err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			osRemoveAll(target)
+		}
+	}()
+
+	lt := LinearTargetPMem(device, deviceSize, deviceStart)
+	linearDeviceName := fmt.Sprintf(linearDeviceFmt, device, deviceStart, deviceSize)
+
+	var dmpath string
+	dmpath, err = dm.CreateDevice(linearDeviceName, dm.CreateReadOnly, []dm.Target{lt})
+	if err != nil {
+		return errors.Wrapf(err, "failed to create dm-linear target: pmem device: %d, offset: %d", device, deviceStart)
+	}
+
+	defer func() {
+		if err != nil {
+			dm.RemoveDevice(linearDeviceName)
+		}
+	}()
+
+	mountFlags := uintptr(unix.MS_RDONLY)
+	if err := unixMount(dmpath, target, "ext4", mountFlags, "noload"); err != nil {
+		return errors.Wrapf(err, "failed to mount dm-linear target %s onto %s", dmpath, target)
+	}
+	return nil
+}
+
+// UnmountDM unmounts `target` and removes associated dm-linear block device
+func UnmountDM(ctx context.Context, deviceNumber uint32, deviceStart, deviceSize int64, target string) (err error) {
+	_, span := trace.StartSpan(ctx, "pmem::UnmountDM")
+	defer span.End()
+	defer func() { oc.SetSpanStatus(span, err) }()
+
+	span.AddAttributes(
+		trace.Int64Attribute("device", int64(deviceNumber)),
+		trace.Int64Attribute("deviceStart", deviceStart),
+		trace.Int64Attribute("deviceSize", deviceSize),
+		trace.StringAttribute("target", target))
+
+	if err := storage.UnmountPath(ctx, target, true); err != nil {
+		return err
+	}
+
+	linearDeviceName := fmt.Sprintf(linearDeviceFmt, deviceNumber, deviceStart, deviceSize)
+	if err := dm.RemoveDevice(linearDeviceName); err != nil {
+		return errors.Wrapf(err, "failed to remove dm-linear target %s", linearDeviceName)
+	}
+	return nil
+}
+
+// LinearTargetPMem returns VPMEM specific dm-linear target
+func LinearTargetPMem(deviceNumber uint32, lengthInBytes, startInBytes int64) dm.Target {
+	startBlock := startInBytes / dm.BlockSize
+	sizeInBlocks := lengthInBytes / dm.BlockSize
+	devicePath := fmt.Sprintf(pMemFmt, deviceNumber)
```

Comment:
I think it makes more sense to have the caller of this function pass in the `devicePath` to use rather than construct one here with a constant fmt. And in that case, I think this function would make more sense in the `devicemapper` package than here.

Reply:
or maybe we don't need it at all. I just wanted to hide the block calculations